vpr/src/route/connection_router.cpp

```diff
@@ -5,6 +5,40 @@
 #include "bucket.h"
 #include "rr_graph_fwd.h"
 
+static inline bool has_path_to_sink(const t_rr_graph_view& rr_nodes,
```

Comment:
When inter-layer connections are solely from OPINs, there's no need to push back the routing path for a connection on another layer into the heap. This function determines whether the node, about to be pushed back into the heap, has an actual path to the sink. For instance, if inter-layer connections are only from OPINs and the node under consideration is of type CHANX/Y, it would only return true if that node is on the same layer as SINK.